daily_milestone.go

```diff
@@ -158,34 +177,42 @@ func getMilestones(baseURL string, token string, projectID string) ([]simpleMile
 func createMilestoneData(advance int, timeInterval string) []simpleMilestone {
 	today := time.Now().Local()
 	list := []simpleMilestone{}
-	for i := 0; i < advance; i++ {
-		if timeInterval == "daily" {
+
+	switch {
+	case timeInterval == "daily":
+		for i := 0; i < advance; i++ {
 			date := today.AddDate(0, 0, i).Format("2006-01-02")
 			milestone := simpleMilestone{}
 			milestone.Title = date
 			milestone.DueDate = date
 			list = append(list, milestone)
-		} else if timeInterval == "weekly" {
-			today = today.AddDate(0, 0, 7)
-			year, week := today.ISOWeek()
-			milestoneYear := strconv.Itoa(year)
-			milestoneWeek := strconv.Itoa(week)
+		}
+	case timeInterval == "weekly":
+		year, week, lastDay := lastDayWeek(today)
```

Comment:
Couldn't that be split up? What's your reasoning to return year and week too?

Reply:
year and week here are for the first iteration after that year and week are used from within the loop

Reply:
ok